manywheel/build_rocm.sh

```diff
@@ -230,26 +237,31 @@ do
 done
 
 DEPS_LIST=(
-    ${ROCM_SO_PATHS[*]}
-    ${OS_SO_PATHS[*]}
+    "${ROCM_SO_PATHS[*]}"
 )
-
+if [[ "$BUILD_LIGHTWEIGHT" != "1" ]]; then
+    DEPS_LIST+=("${OS_SO_PATHS[*]}")
+fi
 DEPS_SONAME=(
-    ${ROCM_SO_FILES[*]}
-    ${OS_SO_FILES[*]}
+    "${ROCM_SO_FILES[*]}"
 )
+if [[ "$BUILD_LIGHTWEIGHT" != "1" ]]; then
+    DEPS_SONAME+=("${OS_SO_FILES[*]}")
+fi
 
-DEPS_AUX_SRCLIST=(
-    "${ROCBLAS_LIB_FILES[@]/#/$ROCBLAS_LIB_SRC/}"
-    "${HIPBLASLT_LIB_FILES[@]/#/$HIPBLASLT_LIB_SRC/}"
-    "/opt/amdgpu/share/libdrm/amdgpu.ids"
-)
+DEPS_AUX_SRCLIST=()
+if [[ "$BUILD_LIGHTWEIGHT" != "1" ]]; then
+    DEPS_AUX_SRCLIST+=("${ROCBLAS_LIB_FILES[@]/#/$ROCBLAS_LIB_SRC}")
+    DEPS_AUX_SRCLIST+=("${HIPBLASLT_LIB_FILES[@]/#/$HIPBLASLT_LIB_SRC}")
+    DEPS_AUX_SRCLIST+=("/opt/amdgpu/share/libdrm/amdgpu.ids")
+fi
 
-DEPS_AUX_DSTLIST=(
-    "${ROCBLAS_LIB_FILES[@]/#/$ROCBLAS_LIB_DST/}"
-    "${HIPBLASLT_LIB_FILES[@]/#/$HIPBLASLT_LIB_DST/}"
-    "share/libdrm/amdgpu.ids"
-)
+DEPS_AUX_DSTLIST=()
+if [[ "$BUILD_LIGHTWEIGHT" != "1" ]]; then
+    DEPS_AUX_DSTLIST+=("${ROCBLAS_LIB_FILES[@]/#/$ROCBLAS_LIB_DST}")
+    DEPS_AUX_DSTLIST+=("${HIPBLASLT_LIB_FILES[@]/#/$HIPBLASLT_LIB_DST}")
+    DEPS_AUX_DSTLIST+=("share/libdrm/amdgpu.ids")
+fi
 
 if [[ $ROCM_INT -ge 50500 ]]; then
```

Comment:
I believe we have to skip this whole block of code which updates `DEPS_AUX_SRCLIST` and `DEPS_AUX_DSTLIST` and keep it under `if [[ "$BUILD_LIGHTWEIGHT" != "1" ]]`, since these files are not required for lightweight whls.

cc @jithunnair-amd

Reply:
It removes below files from whl

  inflating: torch/share/rccl/msccl-algorithms/allgather-8n-0-8kb.xml
  inflating: torch/share/rccl/msccl-algorithms/allgather-8n-128kb-1mb.xml
  inflating: torch/share/rccl/msccl-algorithms/allgather-8n-1mb-40mb.xml
  inflating: torch/share/rccl/msccl-algorithms/allgather-8n-8kb-128kb.xml
  inflating: torch/share/rccl/msccl-algorithms/allgather-allpairs-16n-16tb.xml
  inflating: torch/share/rccl/msccl-algorithms/allreduce-1step-4n-ll-1pass.xml
  inflating: torch/share/rccl/msccl-algorithms/allreduce-allpairs-16n-16tb.xml
  inflating: torch/share/rccl/msccl-algorithms/allreduce-allpairs-16n-32tb.xml
  inflating: torch/share/rccl/msccl-algorithms/allreduce-allpairs-8n-ll-1pass-op.xml
  inflating: torch/share/rccl/msccl-algorithms/allreduce-allpairs-8n-ll-1pass.xml
  inflating: torch/share/rccl/msccl-algorithms/allreduce-allpairs-8n-ll-32tb-op.xml
  inflating: torch/share/rccl/msccl-algorithms/allreduce-allpairs-8n-ll-32tb.xml
  inflating: torch/share/rccl/msccl-algorithms/allreduce-allpairs-8n-ll-64tb-op.xml
  inflating: torch/share/rccl/msccl-algorithms/allreduce-allpairs-8n-ll-64tb.xml
  inflating: torch/share/rccl/msccl-algorithms/allreduce-allpairs-8n-simple_2.xml
  inflating: torch/share/rccl/msccl-algorithms/allreduce-allpairs-8n-simple-op.xml
  inflating: torch/share/rccl/msccl-algorithms/allreduce-allpairs-8n-simple.xml
  inflating: torch/share/rccl/msccl-algorithms/alltoall-8n-0-9kb.xml
  inflating: torch/share/rccl/msccl-algorithms/alltoall-8n-190kb-512kb.xml
  inflating: torch/share/rccl/msccl-algorithms/alltoall-8n-512kb-7mb.xml
  inflating: torch/share/rccl/msccl-algorithms/alltoall-8n-7mb-43mb.xml
  inflating: torch/share/rccl/msccl-algorithms/alltoall-8n-9kb-190kb.xml

Reply:
inflating: torch/share/miopen/db/gfx90a68.db.txt
  inflating: torch/share/miopen/db/gfx90a68.HIP.fdb.txt
  inflating: torch/share/miopen/db/gfx90a6e.db.txt
  inflating: torch/share/miopen/db/gfx90a6e.HIP.fdb.txt
  inflating: torch/share/miopen/db/gfx90a_ConvHipIgemmGroupFwdXdlops_decoder.ktn.model
  inflating: torch/share/miopen/db/gfx90a_ConvHipIgemmGroupFwdXdlops_encoder.ktn.model
  inflating: torch/share/miopen/db/gfx90a_ConvHipIgemmGroupFwdXdlops_metadata.ktn.model
  inflating: torch/share/miopen/db/gfx90a_ConvHipIgemmGroupXdlops_decoder.ktn.model
  inflating: torch/share/miopen/db/gfx90a_ConvHipIgemmGroupXdlops_encoder.ktn.model
  inflating: torch/share/miopen/db/gfx90a_ConvHipIgemmGroupXdlops_metadata.ktn.model
  inflating: torch/share/miopen/db/gfx90a_metadata.tn.model
  inflating: torch/share/miopen/db/gfx90a.tn.model
  inflating: torch/share/miopen/db/gfx942130.db.txt
  inflating: torch/share/miopen/db/gfx942130.HIP.fdb.txt
  inflating: torch/share/miopen/db/gfx942_ConvHipIgemmGroupXdlops_decoder.ktn.model
  inflating: torch/share/miopen/db/gfx942_ConvHipIgemmGroupXdlops_encoder.ktn.model
  inflating: torch/share/miopen/db/gfx942_ConvHipIgemmGroupXdlops_metadata.ktn.model
  inflating: torch/share/miopen/db/gfx942_metadata.tn.model
  inflating: torch/share/miopen/db/gfx942.tn.model

Reply:
To clarify - place lines 252-264 under the if [[ "$BUILD_LIGHTWEIGHT" != "1" ]] ? I may be confused, but I believe we are already doing that.